core/cache.py

```
@@ -0,0 +1,182 @@
+        cache_data = {
+            "cache_key": cache_key,
+            "timestamp": datetime.now().isoformat(),
+            "result": result,
+            "metadata": kwargs
+        }
+        
+        try:
+            with open(cache_file, 'w', encoding='utf-8') as f:
+                json.dump(cache_data, f, indent=2, ensure_ascii=False)
+        except Exception as e:
+            logger.exception(f"Cache write failed for {cache_file}")
```

Ruff (0.15.9)

[error] Undefined name `logger`

(F821)